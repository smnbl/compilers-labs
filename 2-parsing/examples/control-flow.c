int ifs()
{
    if (1 == 2) {
        42;
    }

    if (1 == 2)
        42;
    else
        64;
}

int whiles()
{
    while (1 == 2) {
        42;
    }

    while (1 == 2) ;
}

int returns()
{
    return;
    return 1 + 2;
}
