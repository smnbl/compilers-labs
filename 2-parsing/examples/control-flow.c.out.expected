└── Program
    ├── FuncDecl: returnType = 'int', name = 'ifs'
    │   └── CompoundStmt
    │       ├── IfStmt
    │       │   ├── BinaryOpExpr: op = '=='
    │       │   │   ├── IntLiteral: value = '1'
    │       │   │   └── IntLiteral: value = '2'
    │       │   └── CompoundStmt
    │       │       └── ExprStmt
    │       │           └── IntLiteral: value = '42'
    │       └── IfStmt
    │           ├── BinaryOpExpr: op = '=='
    │           │   ├── IntLiteral: value = '1'
    │           │   └── IntLiteral: value = '2'
    │           ├── ExprStmt
    │           │   └── IntLiteral: value = '42'
    │           └── ExprStmt
    │               └── IntLiteral: value = '64'
    ├── FuncDecl: returnType = 'int', name = 'whiles'
    │   └── CompoundStmt
    │       ├── WhileStmt
    │       │   ├── BinaryOpExpr: op = '=='
    │       │   │   ├── IntLiteral: value = '1'
    │       │   │   └── IntLiteral: value = '2'
    │       │   └── CompoundStmt
    │       │       └── ExprStmt
    │       │           └── IntLiteral: value = '42'
    │       └── WhileStmt
    │           ├── BinaryOpExpr: op = '=='
    │           │   ├── IntLiteral: value = '1'
    │           │   └── IntLiteral: value = '2'
    │           └── EmptyStmt
    └── FuncDecl: returnType = 'int', name = 'returns'
        └── CompoundStmt
            ├── ReturnStmt
            └── ReturnStmt
                └── BinaryOpExpr: op = '+'
                    ├── IntLiteral: value = '1'
                    └── IntLiteral: value = '2'
